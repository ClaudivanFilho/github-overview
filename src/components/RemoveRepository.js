import React, { Component } from 'react'
import PropTypes from 'prop-types'

import ToastContext from '../contexts/ToastContext'
import RepositoryContext from '../contexts/RepositoryContext'
import RemoveIcon from '../images/remove.svg'
import Api from '../api'

class RemoveRepository extends Component {
  static propTypes = {
    repository: PropTypes.object,
    openToast: PropTypes.func,
    refetchRepositories: PropTypes.func,
  }

  handleRemove = (evt) => {
    evt.stopPropagation()
    const [owner, name] = this.props.repository.nameWithOwner.split('/')
    Api.removeRepository(owner, name)
      .then(res => res.json())
      .then(res => {
        if (res.error) {
          this.props.openToast(res.error, true)
        } else {
          this.props.refetchRepositories()
          this.props.openToast(res.message)
        }
      })
      .catch(res => this.props.openToast(res.error, true))
  }

  render() {
    return (
      <img
        className="flex items-center br7 mr3 pointer dim"
        src={RemoveIcon} alt="remove"
        width="20"
        title="remove"
        onClick={this.handleRemove}
      />
    )
  }
}

export default function ComponentWithContext(props) {
  return (
    <ToastContext.Consumer>
      {({ openToast }) => (
        <RepositoryContext.Consumer>
          {({ refetchRepositories }) => (
            <RemoveRepository
              {...props}
              openToast={openToast}
              refetchRepositories={refetchRepositories} />
          )}
        </RepositoryContext.Consumer>
      )}
    </ToastContext.Consumer>
  )
}
