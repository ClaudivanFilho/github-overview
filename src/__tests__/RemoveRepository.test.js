import React from 'react'
import { mount } from 'enzyme'
import toJson from 'enzyme-to-json'

import RemoveRepository from '../components/RemoveRepository'

describe('Test RemoveRepository component', () => {
  let mountedComponent = null

  beforeAll(() => {
    mountedComponent = mount(
      <RemoveRepository />
    )
  })

  it('Renders correctly', () => {
    expect(toJson(mountedComponent)).toMatchSnapshot()
  })

  it('Component should be defined', () => {
    expect(mountedComponent).toBeDefined()
  })

  it('Must find the remove icon', () => {
    expect(mountedComponent.find('img').exists()).toBe(true)
  })
})
