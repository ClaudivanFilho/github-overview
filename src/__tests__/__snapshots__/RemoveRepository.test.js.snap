// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test RemoveRepository component Renders correctly 1`] = `
<ComponentWithContext>
  <RemoveRepository
    openToast={[Function]}
    refetchRepositories={[Function]}
  >
    <img
      alt="remove"
      className="flex items-center br7 mr3 pointer dim"
      onClick={[Function]}
      src="remove.svg"
      title="remove"
      width="20"
    />
  </RemoveRepository>
</ComponentWithContext>
`;
